adapters/tappx/tappx.go

```diff
@@ -136,16 +138,30 @@ func (a *TappxAdapter) buildEndpointURL(params *openrtb_ext.ExtImpTappx, test in
 		}
 	}
 
-	thisURI := host + params.Endpoint + "?tappxkey=" + params.TappxKey
+	thisURI.Path += params.Endpoint
+
+	queryParams := url.Values{}
+
+	if reqKey == "" {
+		return "", &errortypes.BadInput{
```

Comment:
It would be easier to just change the `if` condition on L108 to be:
```
if params.TappxKey == "" {
  return "", &errortypes.BadInput{
    Message: "Tappx key undefined",
  }
}
```

and fail early